url4aac7e6f55_simon0191_tetris3d_tgz-pJ8-Tetris3dJ8/src/Tetris3d.java

```diff
@@ -0,0 +1,459 @@
+        JoglRenderer.verifyOpenGLAvailability();
+        Tetris3d instance = new Tetris3d();
+        instance.appletMode = false;
+        instance.createModel();
+
+        // Create application based GUI
+        JFrame frame;
+        Dimension size;
+
+        instance.createGUI();
+        frame = new JFrame("Tetris 3D");
+        frame.add(instance.canvas, BorderLayout.CENTER);
+        frame.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
+        size = new Dimension(640, 640);
+        frame.setMinimumSize(size);
+        frame.setSize(size);
+        frame.setVisible(true);
+        instance.canvas.requestFocusInWindow();
+    }
+
+//= PROGRAM PART 4/5: JOGL-OPENGL PROCEDURES ================================
+
+    /**
+     * @param gl
+     */
+    private void drawObjectsGL(GL2 gl)
+    {
+        gl.glEnable(gl.GL_DEPTH_TEST);
+
+        gl.glLoadIdentity();
+
+        gl.glPolygonMode(gl.GL_FRONT_AND_BACK, gl.GL_FILL);
+        //corridor.drawGL(gl);
+        
+     // Configure for inside looking
+        gl.glEnable(gl.GL_CULL_FACE);
+        gl.glCullFace(gl.GL_BACK);
+        
+
+        gl.glLineWidth((float)3.0);
+        gl.glBegin(gl.GL_LINES);
+            gl.glColor3d(1, 0, 0);
+            gl.glVertex3d(0, 0, 0);
+            gl.glVertex3d(1, 0, 0);
+
+            gl.glColor3d(0, 1, 0);
+            gl.glVertex3d(0, 0, 0);
+            gl.glVertex3d(0, 1, 0);
+
+            gl.glColor3d(0, 0, 1);
+            gl.glVertex3d(0, 0, 0);
+            gl.glVertex3d(0, 0, 1);
+        gl.glEnd();
+
+        //drawWalls(gl);
+        
+        
+        gl.glRotated(-90, 1, 0, 0);
+        //gl.glTranslated(Shape.BOX_LENGTH/2, -Shape.BOX_LENGTH/2, 0);
+        
+        
+        
+        gl.glEnable(gl.GL_LIGHTING);
+        JoglMaterialRenderer.activate(gl, testMaterial);
+        JoglLightRenderer.activate(gl, light);
+        JoglLightRenderer.activate(gl, light2);
+       
+        ModelRenderer.drawModel(gl, model, camera, qualitySpec);
+        
+        TetrisShapeRenderer.draw(gl, shape, camera, qualitySpec);
+        
+        gl.glDisable(gl.GL_LIGHTING);
+        
+        
+    }
+
+	private void drawWalls(GL2 gl) {
+		// Floor
+		gl.glColor3d(1.0, 0.0, 0.0);
+        
+        gl.glPushMatrix();{
+        	gl.glRotated(90, 1, 0, 0);
+        	gl.glRotated(90, 0, 0, 1);
+        	gl.glTranslated(0, -5*(Shape.BOX_LENGTH)-Shape.SPACE, -8.0*(Shape.BOX_LENGTH+Shape.SPACE)-Shape.SPACE);
+        	WallRenderer.draw(gl, 5, 5, camera, qualitySpec);
+        }gl.glPopMatrix();
+        
+        int width = 5, height = 9;
+        
+        // Walls
+        gl.glPushMatrix();{
+        	WallRenderer.draw(gl, width, height, camera, qualitySpec);
+        }gl.glPopMatrix();
+        
+        gl.glPushMatrix();{
+        	gl.glRotated(-90, 0, 1, 0);
+        	gl.glTranslated(0, 0, -5*(Shape.BOX_LENGTH)-Shape.SPACE);
+        	WallRenderer.draw(gl, width, height, camera, qualitySpec);
+        }gl.glPopMatrix();
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 240:244, 246:250.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.